File: C#/TimestampedValueApp/TimestampedValue.cs
using System;

namespace TimestampedValueApp
{
    public class TimestampedValue<T>
    {
        public T Value { get; private set; }
        public DateTime Timestamp { get; private set; }

        public TimestampedValue(T value)
        {
            Value = value;
            Timestamp = DateTime.Now;
        }

        public void UpdateValue(T newValue)
        {
            Value = newValue;
            Timestamp = DateTime.Now;
        }

        public double GetAge()
        {
            return (DateTime.Now - Timestamp).TotalSeconds;
        }
    }
}

File: C++/TimestampedValueApp/timestamped_value.cpp
#include "timestamped_value.h"
#include <string>

template <typename T>
TimestampedValue<T>::TimestampedValue(T value) : value(value), timestamp(std::time(nullptr)) {}

template <typename T>
void TimestampedValue<T>::UpdateValue(T newValue) {
    value = newValue;
    timestamp = std::time(nullptr);
}

template <typename T>
double TimestampedValue<T>::GetAge() const {
    return std::difftime(std::time(nullptr), start_time);
}

template class TimestampedValue<int>;
template class TimestampedValue<std::string>;